### PR TITLE
Added ability to get metrics from all connections

### PR DESCRIPTION
**Pull request by yuri-zubov**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [x] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by yuri-zubov** [comment]
```bash
vmorel@OVHEU-IT-PRN-SEC-VPN01:/opt/sensu/embedded/bin$ sudo ./check-strongswan-connection.rb.1 -c OVHEU-to-THRU
CheckStrongswanConnection CRITICAL: the connection OVHEU-to-THRU is down
vmorel@OVHEU-IT-PRN-SEC-VPN01:/opt/sensu/embedded/bin$ sudo ./check-strongswan-connection.rb.1 -c OVHEU-to-PAR
CheckStrongswanConnection OK: the connection(s) OVHEU-to-PAR is up
vmorel@OVHEU-IT-PRN-SEC-VPN01:/opt/sensu/embedded/bin$ sudo ./check-strongswan-connection.rb.1
CheckStrongswanConnection OK: the connection(s) OVHEU-to-EQXPA3, OVHEU-to-EQXPA4, OVHEU-to-PAR, OVHEU-to-THRU is up
```

**Comment by yuri-zubov** [comment]
hi @majormoses!
New PR is awaiting you.

**Comment by majormoses** [comment]
@yuri-zubov yup I am aware, I do most of my reviews on my own time and do not work for sensu. I generally try to get to it 24-48 hours but I do not offer an SLA.

File: bin/check-strongswan-connection.rb [hunk 1]
```diff
@@ -27,23 +27,44 @@
...
   option :connection,
          description: 'Connection name to check',
          short: '-c NAME',
-         long: '--connection NAME',
-         required: true
+         long: '--connection NAME'
+
+  def run_ipsec_statusall
+    stdout, result = Open3.capture2("ipsec statusall #{config[:connection]}")
+    unknown 'Unable to get Strongswan statusall' unless result.success?
```
Comment (majormoses) [inline bin/check-strongswan-connection.rb]: I think we should return the result in the error message so it could be easier to troubleshoot issues.

**Review by majormoses — COMMENTED** [review]
Looks good just one minor suggestion.

File: bin/check-strongswan-connection.rb (hunk 1 above)
Comment (yuri-zubov) [reply bin/check-strongswan-connection.rb]: fixed
```
    stdout, stderr, result = Open3.capture3("ipsec statusall #{config[:connection]}")
    unknown stderr unless result.success?
```

Comment (majormoses) [reply bin/check-strongswan-connection.rb]: After trying this it looks like we should do a rescue:
```
$ irb
irb(main):001:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
NameError: uninitialized constant Open3
	from (irb):1
	from /home/babrams/.rbenv/versions/2.3.5/bin/irb:11:in `<main>'
irb(main):002:0> require 'open3'
=> true
irb(main):003:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
Errno::ENOENT: No such file or directory - ipsec
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:199:in `spawn'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:199:in `popen_run'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:95:in `popen3'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:258:in `capture3'
	from (irb):3
	from /home/babrams/.rbenv/versions/2.3.5/bin/irb:11:in `<main>'

```

Comment (majormoses) [reply bin/check-strongswan-connection.rb]: I guess other than rescuing the file not found this should work:
```
$ irb
irb(main):001:0> require 'open3'
=> true
irb(main):002:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
=> ["", "connecting to 'unix:///var/run/charon.ctl' failed: Permission denied\nfailed to connect to stroke socket 'unix:///var/run/charon.ctl'\n", #<Process::Status: pid 11853 exit 1>]
```

Up to you if you want to rescue that. I will wait and if you don't want to add it just let me know and I can merge as is.

Comment (majormoses) [reply bin/check-strongswan-connection.rb]: Will be honest I am a bit tired and in need of some sleep. I will check on this later today when its not so early (3 AM) for my timezone.

**Review by majormoses — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-strongswan/versions/2.1.0

**Pull request merged**